### PR TITLE
CDAP-11615 macro enable table schema

### PR DESCRIPTION
**Pull request by albertshau**
Enabling macros for the schema property of tables.

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/HYP-BAD303

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java [hunk 1]
```diff
@@ -74,6 +75,12 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   }
 
   @Override
+  protected boolean shouldSkipCreateAtConfigure() {
+    return tableSinkConfig.containsMacro(Properties.Table.PROPERTY_SCHEMA) ||
```
Comment (gokulavasan) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java:79]: How does these two values being macros affect the creation of the dataset during configure time? Doesn't it only affect the way StructuredRecord->Put transform happens?

Comment (albertshau) [reply core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java:79]: if we create them at configure time, schema will be unavailable and they will not be explorable

Comment (gokulavasan) [reply core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java:79]: Oh yes ok, it is a CDAP table not HBase table. Thanks for the clarification!

**Comment by gokulavasan** [comment]
LGTM 👍

**Pull request merged**